pkg/util/mocks/mock_types.go

```diff
@@ -0,0 +1,138 @@
+package mocks
+
+import (
+	"fmt"
+
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/secret"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/authtypes"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type MockSecretGetUpdateCreateDeleter struct {
+	secrets map[client.ObjectKey]corev1.Secret
+}
+
+func NewMockedSecretGetUpdateCreateDeleter(secrets ...corev1.Secret) secret.GetUpdateCreateDeleter {
```

Comment:
is there a mock lying around somewhere which we can re-use instead of creating a new one?